packages/account-v2/src/modules/FinancialAssessment/__tests__/FinancialAssessmentForm.spec.tsx

```diff
@@ -0,0 +1,230 @@
+        expect(screen.getByText('Let’s continue with providing proofs of address and identity.')).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Continue' })).toBeInTheDocument();
+    });
+
+    it('render Demo Message if account virtual', () => {
+        (useActiveTradingAccount as jest.Mock).mockReturnValue({
+            data: {
+                is_virtual: true,
+            },
+        });
+        render(<FinancialAssessmentForm />);
+        expect(screen.getByText('DemoMessage')).toBeInTheDocument();
+    });
+
+    it('show loader when api call in progress', () => {
+        (useFinancialAssessment as jest.Mock).mockReturnValue({
+            data: {},
+            error: undefined,
+            isLoading: true,
+            mutation: {
+                error: undefined,
+                isLoading: false,
+                isSuccess: false,
+            },
+            update: jest.fn(),
+        });
+        render(<FinancialAssessmentForm />);
+        expect(screen.getByText('Loading...')).toBeInTheDocument();
+    });
+
+    it('render Error Message on fetch error', () => {
+        (useFinancialAssessment as jest.Mock).mockReturnValue({
+            data: {},
+            error: {
+                error: {
+                    message: 'Error Loading API',
+                },
+            },
+            isLoading: false,
+            mutation: {
+                error: undefined,
+                isLoading: false,
+                isSuccess: false,
+            },
+            update: jest.fn(),
+        });
+        render(<FinancialAssessmentForm />);
+        expect(screen.getByText('Error Loading API')).toBeInTheDocument();
+    });
+
+    it('render Update Error Message on update error', () => {
+        (useFinancialAssessment as jest.Mock).mockReturnValue({
+            data: {},
+            error: undefined,
+            isLoading: false,
+            mutation: {
+                error: {
+                    error: {
+                        message: 'Error Updating API',
+                    },
+                },
+                isLoading: false,
+                isSuccess: false,
+            },
+            update: jest.fn(),
+        });
+        render(<FinancialAssessmentForm />);
+        expect(screen.getByText('Error Updating API')).toBeInTheDocument();
+    });
+
+    it('renders Financial Assessment form', () => {
+        render(<FinancialAssessmentForm />);
+        expect(screen.getByText('Financial information')).toBeInTheDocument();
+        expect(screen.getByText('(All fields are required)')).toBeInTheDocument();
+        expect(screen.getByRole('combobox', { name: 'Source of income' })).toBeInTheDocument();
+        expect(screen.getByRole('combobox', { name: 'Source of wealth' })).toBeInTheDocument();
+    });
+
+    it('calls update function with valid values on form submission', async () => {
+        render(<FinancialAssessmentForm />);
+
+        act(() => {
```

Comment:
we don't need to wrap user events in act, it is wrapped under the hood. it was not working without act?